### DISCUSSION TITLE
add Kandinsky 2.0 - the first multilingual text2image model

### DISCUSSION THREAD

**Discussion by 0-NiK-0** [hlky/stable-diffusion-webui#1761] [Feature suggestions]
Ability to write Prompt in more than 100 languages.

Kandinsky 2.0
https://github.com/ai-forever/Kandinsky-2.0
https://huggingface.co/sberbank-ai/Kandinsky_2.0
https://fusionbrain.ai/diffusion

Model architecture:
It is a latent diffusion model with two multilingual text encoders:

mCLIP-XLMR 560M parameters
mT5-encoder-small 146M parameters
These encoders and multilingual training datasets unveil the real multilingual text-to-image generation experience!

Kandinsky 2.0 was trained on a large 1B multilingual set, including samples that we used to train Kandinsky.

In terms of diffusion architecture Kandinsky 2.0 implements UNet with 1.2B parameters.

Kandinsky 2.0 architecture overview:
<img width="1428" alt="NatallE" src="https://user-images.githubusercontent.com/126181396/227786016-49e2a1d3-b2dc-4c55-a645-198b7989cef0.png">

**Comment by 0-NiK-0** [comment]
Kandinsky 2.1 inherits best practicies from Dall-E 2 and Latent diffucion, while introducing some new ideas.

As text and image encoder it uses CLIP model and diffusion image prior (mapping) between latent spaces of CLIP modalities. This approach increases the visual performance of the model and unveils new horizons in blending images and text-guided image manipulation.

For diffusion mapping of latent spaces we use transformer with num_layers=20, num_heads=32 and hidden_size=2048.

Other architecture parts:

Text encoder (XLM-Roberta-Large-Vit-L-14) - 560M
Diffusion Image Prior — 1B
CLIP image encoder (ViT-L/14) - 427M
Latent Diffusion U-Net - 1.22B
MoVQ encoder/decoder - 67M
Kandinsky 2.1 was trained on a large-scale image-text dataset LAION HighRes and fine-tuned on our internal datasets.
<img width="730" alt="kandinsky21" src="https://user-images.githubusercontent.com/126181396/228551693-6d975cd5-9243-48e0-a7b3-4ef09e79d540.png">

**Comment by ke1ne** [comment]
1. This guy looks like a spammer who is pushing the same message into all Stable Diffusion related discussion.
2. "Kandinsky" model team is a russian bank. The Russia is waing a genocidal war in Ukraine, killing, raping and torturing Ukrainian people. The International Criminal Court issued an arrest warrant for Russian President Putin for forced deporting Ukrainian children to Russia. It's ridiculus and not ethical to support russian products!

**Reply by FurkanGozukara** [reply]
i dont think so using a public model for free would benefit russia a bit

**Reply by ke1ne** [reply]
@FurkanGozukara Ideological benefit, like popularity, is also benefit
